src/java.base/share/classes/sun/security/ssl/CertificateAuthoritiesExtension.java

```diff
@@ -277,7 +283,11 @@ public void consume(ConnectionContext context,
                     new CertificateAuthoritiesSpec(shc, buffer);
 
             // Update the context.
-            shc.peerSupportedAuthorities = spec.getAuthorities();
+            try {
+                shc.peerSupportedAuthorities = spec.getAuthorities();
+            } catch (IllegalArgumentException iae) {
+                shc.conContext.fatal(Alert.DECODE_ERROR, "X500Principal could not be parsed");
```

Comment:
To easy debugging, I may use the exception with fatal(Alert alert, String diagnostic, Throwable cause).  Considering this point, I may prefer to throw SSLException in getAuthorities() method.

```
X500Principal[] getAuthorities() throws SSLException {
...
try {
    shc.peerSupportedAuthorities = spec.getAuthorities();
} catch (SSLException ssle) {
    shc.conContext.fatal(Alert.DECODE_ERROR, "Cannot parse peer supported authorities", ssle);
}
```